hazelcast/src/main/java/com/hazelcast/nio/serialization/PortableReader.java

```diff
@@ -36,165 +43,285 @@
      * @param fieldName name of the field (does not support nested paths)
      * @return true if field exist in this class.
      */
-    boolean hasField(String fieldName);
+    boolean hasField(@Nonnull String fieldName);
 
     /**
      * @return set of field names on this portable class
      */
+    @Nonnull
     Set<String> getFieldNames();
 
     /**
      * @param fieldName name of the field
      * @return field type of given fieldName
```

Comment:
nit: javadoc doesn't mention the fact that this method throws `IllegalArgumentException` when the field is not present